src/test/java/io/github/christophermanahan/carnitas/RouterTest.java

```diff
@@ -5,82 +5,100 @@
         Router router = new Router()
           .post("/simple_post", handler);
 
         HTTPResponse response = router.handle(request);
 
-        assertArrayEquals(handler.apply(request).serialize(), response.serialize());
+        assertTrue(handler.apply(request).equals(response));
     }
 
     @Test
-    void itProcessesARequestIntoANotFoundResponseIfTheRouteHasNotBeenAdded() {
-        Function<HTTPRequest, HTTPResponse> handler = (HTTPRequest request) -> new HTTPResponse(HTTPResponse.Status.OK);
-        HTTPRequest request = new HTTPRequest(HTTPRequest.Method.POST, "/simple_post");
+    void itProcessesARequestIntoAResponseIfMultipleRoutesHaveBeenAdded() {
+        Function<HTTPRequest, HTTPResponse> handler = (HTTPRequest request) -> new ResponseBuilder()
+          .setStatus(HTTPResponse.Status.OK)
+          .get();
+        HTTPRequest request = new HTTPRequest(HTTPRequest.Method.GET, "simple_get_again");
         Router router = new Router()
-          .get( "/simple_get", handler);
+          .get( "/simple_get", handler)
+          .get("simple_get_again", handler);
 
         HTTPResponse response = router.handle(request);
 
-        assertArrayEquals(new HTTPResponse(HTTPResponse.Status.NOT_FOUND).serialize(), response.serialize());
+        assertTrue(handler.apply(request).equals(response));
     }
 
     @Test
-    void itProcessesAGETRequestIntoAResponseIfMultipleRoutesHaveBeenAdded() {
-        Function<HTTPRequest, HTTPResponse> handler = (HTTPRequest request) -> new HTTPResponse(HTTPResponse.Status.OK);
-        HTTPRequest request = new HTTPRequest(HTTPRequest.Method.GET, "simple_get_again");
+    void itProcessesARequestIntoANotFoundResponseIfTheRouteHasNotBeenAdded() {
+        Function<HTTPRequest, HTTPResponse> handler = (HTTPRequest request) -> new ResponseBuilder()
+          .setStatus(HTTPResponse.Status.OK)
+          .get();
+        HTTPRequest request = new HTTPRequest(HTTPRequest.Method.POST, "/simple_post");
         Router router = new Router()
-          .get( "/simple_get", handler)
-          .get("simple_get_again", handler);
+          .get( "/simple_get", handler);
 
         HTTPResponse response = router.handle(request);
 
-        assertArrayEquals(new HTTPResponse(HTTPResponse.Status.OK).serialize(), response.serialize());
-
+        HTTPResponse expectedResponse = new ResponseBuilder()
+          .setStatus(HTTPResponse.Status.NOT_FOUND)
+          .addHeader(Headers.contentLength(0))
+          .get();
+        assertTrue(expectedResponse.equals(response));
     }
 
     @Test
-    void itProcessesAPOSTRequestIntoAResponseIfTheRouteHasBeenAddedToDifferentMethodsThanTheRequestMethod() {
-        Function<HTTPRequest, HTTPResponse> handler = (HTTPRequest request) -> new HTTPResponse(HTTPResponse.Status.CREATED);
+    void itProcessesARequestIntoAMethodNotAllowedResponseIfTheRouteHasBeenAddedToDifferentMethodsThanTheRequestMethod() {
```

Comment:
I might rephrase to _it does not allow requests to routes that cannot respond to the request method_ or something like that.

I like to keep the descriptions closer to the domain vs describing the implementation. Does that make sense?